test/rmui_bundle_suggestors/dart_script_adder_test.dart

```diff
@@ -0,0 +1,293 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/rmui_bundle_suggestors/constants.dart';
+import 'package:over_react_codemod/src/rmui_bundle_suggestors/dart_script_adder.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+import 'html_script_adder_test.dart';
+
+main() {
+  group('DartScriptAdder', () {
+    group('add prod script', () => _dartScriptAdderTests(isProd: true));
+
+    group('add non-prod script', () => _dartScriptAdderTests(isProd: false));
+  });
+}
+
+void _dartScriptAdderTests({bool isProd = true}) {
+  final expectedAddedScript = isProd ? rmuiBundleProd : rmuiBundleDev;
+  final testSuggestor =
+      getSuggestorTester(DartScriptAdder(expectedAddedScript, isProd));
```

Comment:
Are there tests in this and the HTML suite that verify it doesn't add the prod RMUI bundle when there are dev React files, and doesn't add the dev RMUI bundle when there are prod React files?

One easy way to test this could be to run the two suggestors together (since they'll never really be run individually) and just verify the output for each case, as opposed to testing them separately and making sure they behave correctly in all scenarios. That also would provide test coverage that they work together well.

```dart
  final testSuggestor = getSuggestorTester(aggregate([
    DartScriptAdder(rmuiBundleProd, true),
    DartScriptAdder(rmuiBundleDev, false)
  ]));
}

Reply:
Since you're out this week, I pushed up a branch with those proposed changes: https://github.com/Workiva/over_react_codemod/pull/127 Lemme know what you think!

Reply:
@sydneyjodon-wk Since you're out this week, I pushed up a branch with those proposed changes: https://github.com/Workiva/over_react_codemod/pull/127 Lemme know what you think!

Reply:
> so feel free to make commits / changes to any of my open PRs in the meantime

Just saw this, I guess I'll just push up to this branch and hope you're okay with it! 😄

Reply:
Addressed in https://github.com/Workiva/over_react_codemod/pull/126/commits/22ad72550e1859449ce32a1789d87a88f6d2514b?diff=unified&w=1